core/templates/rid_owner.h

```diff
@@ -290,9 +381,9 @@ class RID_Alloc : public RID_AllocBase {
 			mutex.lock();
 		}
 		for (size_t i = 0; i < max_alloc; i++) {
-			uint64_t validator = chunks[i / elements_in_chunk][i % elements_in_chunk].validator;
+			uint32_t validator = chunks[i / elements_in_chunk][i % elements_in_chunk].validator;
 			if (validator != 0xFFFFFFFF) {
-				p_owned->push_back(_make_from_id((validator << 32) | i));
+				p_owned->push_back(_make_from_id(((uint64_t)validator << 32) | i));
```

Comment:
Confusing.

It's not clear whether it is `uint64_t( validator ) << 32` (good) or `uint64_t( validator << 32 )` (bad).
`i` is also size_t which is uint32_t in arm32 and could cause issues. Better to cast it too:

```cpp
p_owned->push_back(_make_from_id( uint64_t( validator ) << 32) | uint64_t( i ) );
```

Reply:
I'm just keeping these two lines untouched. I had stylistic reasons, but those changes aren't part of the scope of this PR, after all.